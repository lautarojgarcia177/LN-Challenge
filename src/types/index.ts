// Interfaces generadas con app.quicktype.io

export interface Article {
  _id: string;
  display_date: string;
  headlines: Headlines;
  promo_items?: PromoItems;
  subtype: string;
  taxonomy: Taxonomy;
  website_url: string;
}

export interface Headlines {
  basic: string;
}

export interface PromoItems {
  basic: Basic;
}

export interface Basic {
  resized_urls?: ResizedURL[];
  subtitle?: string;
  type: Type;
  url?: string;
}

export interface ResizedURL {
  option: Option;
  resizedUrl: string;
}

export interface Option {
  media: Media;
}

export enum Media {
  MinWidth20Em = "(min-width: 20em)",
  MinWidth48Em = "(min-width: 48em)",
  MinWidth64Em = "(min-width: 64em)",
}

export enum Type {
  Image = "image",
  Video = "video",
}

export interface Tag {
  slug: string;
  text: string;
}

export interface Taxonomy {
  tags: Tag[];
}

export interface TagCount extends Tag {
  count: number;
}
