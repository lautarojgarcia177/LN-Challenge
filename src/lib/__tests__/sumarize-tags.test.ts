import { Article } from "@/types";
import sumarizeTags from "../sumarize-tags";

describe("sumarizeTags", () => {
  it("retorna un array vacio cuando no hay articulos", () => {
    const result = sumarizeTags([]);
    expect(result).toEqual([]);
  });

  it("summarizes tags correctly", () => {
    const articles = [
      {
        taxonomy: {
          tags: [
            { slug: "tag1", text: "Tag 1" },
            { slug: "tag2", text: "Tag 2" },
          ],
        },
      },
      {
        taxonomy: {
          tags: [
            { slug: "tag1", text: "Tag 1" },
            { slug: "tag3", text: "Tag 3" },
          ],
        },
      },
    ] as Article[];
    const result = sumarizeTags(articles);
    expect(result).toEqual([
      { slug: "tag1", text: "Tag 1", count: 2 },
      { slug: "tag2", text: "Tag 2", count: 1 },
      { slug: "tag3", text: "Tag 3", count: 1 },
    ]);
  });

  it("limita el numero de etiquetas retornadas en base al parametro 'amount'", () => {
    const articles = [
      {
        taxonomy: {
          tags: [
            { slug: "tag1", text: "Tag 1" },
            { slug: "tag2", text: "Tag 2" },
          ],
        },
      },
      {
        taxonomy: {
          tags: [
            { slug: "tag1", text: "Tag 1" },
            { slug: "tag3", text: "Tag 3" },
          ],
        },
      },
      { taxonomy: { tags: [{ slug: "tag4", text: "Tag 4" }] } },
    ] as Article[];
    const result = sumarizeTags(articles, 2);
    expect(result).toEqual([
      { slug: "tag1", text: "Tag 1", count: 2 },
      { slug: "tag2", text: "Tag 2", count: 1 },
    ]);
  });

  it("Maneja los articulos sin tags correctamente", () => {
    const articles = [
      { taxonomy: { tags: [] } },
      { taxonomy: { tags: [{ slug: "tag1", text: "Tag 1" }] } },
    ] as Article[];
    const result = sumarizeTags(articles);
    expect(result).toEqual([{ slug: "tag1", text: "Tag 1", count: 1 }]);
  });

  it("Maneja los articulos con taxonomy undefined correctamente", () => {
    const articles = [
      { taxonomy: undefined },
      { taxonomy: { tags: [{ slug: "tag1", text: "Tag 1" }] } },
    ] as Article[];
    const result = sumarizeTags(articles);
    expect(result).toEqual([{ slug: "tag1", text: "Tag 1", count: 1 }]);
  });

  it("coincide con la snapshot", () => {
    const articles = [
      {
        taxonomy: {
          tags: [
            { slug: "tag1", text: "Tag 1" },
            { slug: "tag2", text: "Tag 2" },
          ],
        },
      },
      {
        taxonomy: {
          tags: [
            { slug: "tag1", text: "Tag 1" },
            { slug: "tag3", text: "Tag 3" },
          ],
        },
      },
    ] as Article[];
    const result = sumarizeTags(articles);
    expect(result).toMatchSnapshot();
  });
});
