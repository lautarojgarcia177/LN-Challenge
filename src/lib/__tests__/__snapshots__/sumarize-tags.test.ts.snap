// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sumarizeTags coincide con la snapshot 1`] = `
[
  {
    "count": 2,
    "slug": "tag1",
    "text": "Tag 1",
  },
  {
    "count": 1,
    "slug": "tag2",
    "text": "Tag 2",
  },
  {
    "count": 1,
    "slug": "tag3",
    "text": "Tag 3",
  },
]
`;
